File: StudentManagement/src/main/java/com/studentmanagement/services/StudentService.java
package com.studentmanagement.services;

import java.util.List;
import java.util.Set;

import com.studentmanagement.entities.Student;
import com.studentmanagement.entities.StudentSubject;

public interface StudentService {

    public List<Student> getAllStudents();

    public Student saveStudent(Student student);

    public Student getStudentById(Long id);

    public Student updateStudent(Student student, long id);

//    public void deleteStudent(Long studentId);
    public void deleteStudent(long id);
//    public List<Student> printbyname(String name);

    public Student createStudent(Student student, Set<StudentSubject> studentSubject);
}

File: studentmanagementapp/src/app/student.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, Subject } from 'rxjs';
import { Student } from './student';
import { Subjects } from './subject';

@Injectable({
  providedIn: 'root'
})
export class StudentService {
  
  private baseURL="http://localhost:8080/api/v1"
  constructor(private httpClient : HttpClient) { }

  getStudentsList(): Observable<Student[]>{
    return this.httpClient.get<Student[]>(`${this.baseURL}/students`)  
  }

  createStudent(student: Student): Observable<any>{
    return this.httpClient.post(`${this.baseURL}/add_student`,student);
  }

  getStudentById(id:number):Observable<Student>{
    return this.httpClient.get<Student>(`${this.baseURL}/students/${id}`)
  }

  updateStudent(id: number, student: Student) : Observable<Student>{
    return this.httpClient.put<Student>(`${this.baseURL}/students/${id}`,student);
  }

  deleteStudent(id: number): Observable<Object>{
    return this.httpClient.delete(`${this.baseURL}/students/${id}`); 
  }
  
   getStudentsSubjectById(id:number):Observable<Student["subjectIds"][]>{
    return this.httpClient.get<Student["subjectIds"][]>(`${this.baseURL}/studentsubject/${id}`);
  }
}
